lib/honeybadger/agent.rb

```diff
@@ -365,6 +365,20 @@ def stop(force = false)
       true
     end
 
+    # Sends event to events backend
+    #
+    # @param event_name [String] a string describing the event
+    # @param payload [Hash] Additional data to be sent with the event as keyword arguments
+    def event(event_type, payload={})
+      ts = DateTime.now.new_offset(0).rfc3339
+      unless payload.kind_of?(Hash)
+        logger.error("Event has non-hash payload")
+        return
+      end
+      log_string = payload.merge({event_type: event_type, ts: ts}).to_json
```

Comment:
Or maybe this:

```suggestion
      log_string = Hash(payload).merge({event_type: event_type, ts: ts}).to_json
```

Reply:
The `event_type` that we are merging here comes from the user, right? So, the argument here is only about `ts`, which I think it's fine to merge, at least for now.

Reply:
Sorry for not commenting sooner.

Although most of the time users will care about the event's timestamp when it is called, there are instances where overriding the timestamp could be useful. For example, if you wanted to log an external event that already has its own timestamp. Therefore, my vote is to allow the user-supplied payload to take precedence.

Reply:
Good point, no objection from my part.
Although, does this mean that internally we have keep an additional timestamp field (such as `created_at`)?

Reply:
> The `event_type` that we are merging here comes from the user, right? So, the argument here is only about `ts`, which I think it's fine to merge, at least for now.

Yeah, that seems right, although it still might make sense for `event_type` in the payload to take precedence.

> Good point, no objection from my part. Although, does this mean that internally we keep an additional timestamp field (such as `created_at`)?

I don't think we need an additional timestamp. (Someone correct me if I'm wrong.)

Reply:
We do keep track of when the event is ingested, which we use internally for bookkeeping.

Reply:
@joshuap @subzero10 @rabidpraxis I've reversed the merge order and use the Hash constructor now. Does this look okay now?

Reply:
LGTM!